metricbeat/module/mongodb/metrics/data.go

```diff
@@ -0,0 +1,167 @@
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package metrics
+
+import (
+	s "github.com/elastic/beats/libbeat/common/schema"
+	c "github.com/elastic/beats/libbeat/common/schema/mapstriface"
+)
+
+var schema = s.Schema{
+	"commands": c.Dict("metrics.commands", s.Schema{
+		"is_self":                 c.Dict("_isSelf", commandSchema),
+		"aggregate":               c.Dict("aggregate", commandSchema),
+		"build_info":              c.Dict("buildInfo", commandSchema),
+		"coll_stats":              c.Dict("collStats", commandSchema),
+		"connection_pool_stats":   c.Dict("connPoolStats", commandSchema),
+		"count":                   c.Dict("count", commandSchema),
+		"db_stats":                c.Dict("dbStats", commandSchema),
+		"distinct":                c.Dict("distinct", commandSchema),
+		"find":                    c.Dict("find", commandSchema),
+		"get_cmd_line_opts":       c.Dict("getCmdLineOpts", commandSchema),
+		"get_last_error":          c.Dict("getLastError", commandSchema),
+		"get_log":                 c.Dict("getLog", commandSchema),
+		"get_more":                c.Dict("getMore", commandSchema),
+		"get_parameter":           c.Dict("getParameter", commandSchema),
+		"host_info":               c.Dict("hostInfo", commandSchema),
+		"insert":                  c.Dict("insert", commandSchema),
+		"is_master":               c.Dict("isMaster", commandSchema),
+		"last_collections":        c.Dict("listCollections", commandSchema),
+		"last_commands":           c.Dict("listCommands", commandSchema),
+		"list_databased":          c.Dict("listDatabases", commandSchema),
+		"list_indexes":            c.Dict("listIndexes", commandSchema),
+		"ping":                    c.Dict("ping", commandSchema),
+		"profile":                 c.Dict("profile", commandSchema),
+		"replset_get_rbid":        c.Dict("replSetGetRBID", commandSchema),
+		"replset_get_status":      c.Dict("replSetGetStatus", commandSchema),
+		"replset_heartbeat":       c.Dict("replSetHeartbeat", commandSchema),
+		"replset_update_position": c.Dict("replSetUpdatePosition", commandSchema),
+		"server_status":           c.Dict("serverStatus", commandSchema),
+		"update":                  c.Dict("update", commandSchema),
+		"whatsmyuri":              c.Dict("whatsmyuri", commandSchema),
```

Comment:
As we are collecting metrics about each one of the commands here, maybe we want to keep the commands as field names, even if they are camel cased, @ruflin wdyt?

Reply:
Tricky one. I would prefer to keep all names lower case if possible but I understand that it might confuse people that search for a specific command. at the same time I would hope autocomplete and when they type the first characters, it becomes clear.